src/test/java/tz/go/moh/him/hfr/mediator/mock/MockDestination.java

```diff
@@ -0,0 +1,68 @@
+package tz.go.moh.him.hfr.mediator.mock;
+
+import com.google.gson.Gson;
+import org.openhim.mediator.engine.messages.MediatorHTTPRequest;
+import org.openhim.mediator.engine.testing.MockHTTPConnector;
+import tz.go.moh.him.hfr.mediator.domain.HfrRequest;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Represents a mock destination.
+ */
+public class MockDestination extends MockHTTPConnector {
+
+    /**
+     * Initializes a new instance of the {@link MockDestination} class.
+     */
+    public MockDestination() {
+    }
+
+    /**
+     * Gets the response.
+     *
+     * @return Returns the response.
+     */
+    @Override
+    public String getResponse() {
+        return null;
+    }
+
+    /**
+     * Gets the status code.
+     *
+     * @return Returns the status code.
+     */
+    @Override
+    public Integer getStatus() {
+        return 200;
+    }
+
+    /**
+     * Gets the HTTP headers.
+     *
+     * @return Returns the HTTP headers.
+     */
+    @Override
+    public Map<String, String> getHeaders() {
+        return Collections.emptyMap();
+    }
+
+    @Override
+    public void executeOnReceive(MediatorHTTPRequest msg) {
+        System.out.println(msg.getBody());
+
+        Gson gson = new Gson();
+
+        HfrRequest request = gson.fromJson(msg.getBody(), HfrRequest.class);
+
+//        assertEquals(expectedPayload.getPartNum(), dailyStockStatus.getPartNum());
+//        assertEquals(expectedPayload.getDate(), dailyStockStatus.getDate());
+//        assertEquals(expectedPayload.getMonthOfStock(), dailyStockStatus.getMonthOfStock());
+//        assertEquals(expectedPayload.getOnHandQty(), dailyStockStatus.getOnHandQty());
+//        assertEquals(expectedPayload.getOum(), dailyStockStatus.getOum());
+//        assertEquals(expectedPayload.getPartDescription(), dailyStockStatus.getPartDescription());
+//        assertEquals(expectedPayload.getPlant(), dailyStockStatus.getPlant());
```

Comment:
I would suggest you assert on that what was sent by the mediator is what is being received by the mock destination system.

Reply:
You might will also need to add additional unit tests to increase the coverage to at least 60% to be allowed to merge the pull request.

Reply:
@cozej4 I've updated the tests now, the internal IntelliJ coverage report states 74% after my latest commit, so hopefully it's good now :)